chain_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build linux
+// +build linux
+
+package netlink
+
+import (
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestChainAddDel(t *testing.T) {
+	tearDown := setUpNetlinkTest(t)
+	defer tearDown()
+	if err := LinkAdd(&Ifb{LinkAttrs{Name: "foo"}}); err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkAdd(&Ifb{LinkAttrs{Name: "bar"}}); err != nil {
+		t.Fatal(err)
+	}
+	link, err := LinkByName("foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkSetUp(link); err != nil {
+		t.Fatal(err)
+	}
+	redir, err := LinkByName("bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkSetUp(redir); err != nil {
+		t.Fatal(err)
+	}
+	qdisc := &Ingress{
+		QdiscAttrs: QdiscAttrs{
+			LinkIndex: link.Attrs().Index,
+			Handle:    MakeHandle(0xffff, 0),
+			Parent:    HANDLE_INGRESS,
+		},
+	}
+	if err := QdiscAdd(qdisc); err != nil {
+		t.Fatal(err)
+	}
+	qdiscs, err := SafeQdiscList(link)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(qdiscs) != 1 {
+		t.Fatal("Failed to add qdisc")
+	}
+	_, ok := qdiscs[0].(*Ingress)
+	if !ok {
+		t.Fatal("Qdisc is the wrong type")
+	}
+	chainVal := new(uint32)
+	*chainVal = 20
+	chain := NewChain(HANDLE_INGRESS, *chainVal)
+	err = ChainAdd(link, chain)
+	if err != nil {
+		t.Fatal(err)
+	}
+	chains, err := ChainList(link, HANDLE_INGRESS)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(chains) != 1 {
+		t.Fatal("Failed to add chain")
+	}
+	if chains[0].Chain != *chainVal {
+		t.Fatalf("Incorrect chain added")
+	}
+	classId := MakeHandle(1, 1)
+	filter := &U32{
```

Comment:
testing filter with chain attr should be part of filter_test.go IMO

Reply:
done